src/Microsoft.Health.Dicom.SqlServer/Features/ExtendedQueryTag/ExtendedQueryTagDataRowsBuilder.cs

```diff
@@ -52,26 +52,46 @@ internal static class ExtendedQueryTagDataRowsBuilder
             var dateTimeWithUtcRows = new List<InsertDateTimeExtendedQueryTagTableTypeV2Row>();
             var personNameRows = new List<InsertPersonNameExtendedQueryTagTableTypeV1Row>();
 
-            foreach (QueryTag queryTag in queryTags.Where(x => x.IsExtendedQueryTag))
+            foreach (QueryTag queryTag in queryTags.Where(x => x.IsExtendedQueryTag || x.WorkitemQueryTagStoreEntry != null))
+            {
+                if (queryTag.VR == DicomVR.SQ)
+                {
+                    var dicomDatasets = instance.GetSequencePathValues(queryTag.WorkitemQueryTagStoreEntry.PathTags);
+                    foreach (var dataset in dicomDatasets)
```

Comment:
nit:
```suggestion
                    foreach (var dicomItem in dicomDatasets.SelectMany(x => x))
```

Reply:
I dont want to do this since we need the dataset to get the value for that item